app/controllers/api/regions_controller.rb

```diff
@@ -1,4 +1,11 @@
 module Api
   class RegionsController < BaseController
+    private
+
+    def resource_search(id, type, klass)
+      region = MiqRegion.find_by(:region => id)
+      return region unless region.nil?
+      super(id, type, klass)
```

Comment:
Hmm, might be problematic GET /api/regions/:id is now non-deterministic.  I think we need to go all one way or the other, href's generated are using region numbers and not record id's.  

take a look at resource_identifier for automate_workspaces. might be helpful.